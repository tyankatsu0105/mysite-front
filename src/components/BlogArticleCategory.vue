<template>
	<div class="BlogArticleCategory">
		<g-link :to="link">
			<font-awesome-icon
				:icon="{ prefix: 'fa', iconName: 'folder' }"
			/>
			<p class="BlogArticleCategory-CategoryText">
				{{ text }}
			</p>
		</g-link>
	</div>
</template>

<script>
export default {
  props: {
    text: {
      type: String,
      requirer: true
    },
    link: {
      type: String,
      requirer: true
    }
  }
};
</script>

<style scoped lang='scss'>
.BlogArticleCategory {
  a {
    transition: $tst-default;

    &:hover {
      color: $color-accent;
    }
  }

  &-CategoryText {
    display: inline-block;
    margin-left: 5px;
    font-size: 1rem;
  }
}
</style>
