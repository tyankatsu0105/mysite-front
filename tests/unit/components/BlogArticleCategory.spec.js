import { shallowMount } from "@vue/test-utils";
import BlogArticleCategory from "@/components/BlogArticleCategory.vue";

const propsData = {
  text: "カテゴリ"
};

describe("BlogArticleCategory", () => {
  it("render", () => {
    const wrapper = shallowMount(BlogArticleCategory, {
      propsData,
      stubs: ["font-awesome-icon"]
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
