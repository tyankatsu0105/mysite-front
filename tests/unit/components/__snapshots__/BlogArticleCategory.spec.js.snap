// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogArticleCategory render 1`] = `
<div class="BlogArticleCategory">
  <g-link to="hogehoge.com">
    <font-awesome-icon icon="[object Object]"></font-awesome-icon>
    <p class="BlogArticleCategory-CategoryText">
      カテゴリ
    </p>
  </g-link>
</div>
`;
